### PR TITLE
Support whitelist in UI

### PR DESCRIPTION
**Pull request by TahirHameed74**
Support whitelist in UI

### CODE REVIEW

File: hooks/useCreateData.tsx [hunk 1]
```diff
@@ -33,11 +47,26 @@ const useCreateData = () => {
...
   const { prepare } = usePreparePrivyWallet()
 
+  useEffect(() => {
+    const checkIfWhitelisted = async () => {
+      if (window.ethereum) {
+        const provider = new ethers.providers.Web3Provider(window.ethereum)
+        const signer = provider.getSigner()
+        const connectedAddress = await signer.getAddress()
+        const isWhitelistedAddress = whitelistedAddresses.includes(connectedAddress)
+        console.log(`Address: ${connectedAddress}, Is Whitelisted: ${isWhitelistedAddress}`)
```
Comment (sweetmantech) [inline hooks/useCreateData.tsx]: why aren't you using the `useConnectedWallet` hook we've already written?
- `hooks/useConnectedWallet`

File: hooks/useCreateData.tsx [hunk 2]
```diff
@@ -12,12 +13,25 @@ import handleTxError from "@/lib/handleTxError"
...
+const whitelistedAddresses = [
+  "0x2d9cBC4ECFBd1B8F66Aa798FD51585AE058dAA8B",
+  "0x7A631B4a6b1D0a6b1e433d92690d7C8Aba4F23b4",
+  "0xD6C09962E907428112069273d5C0dc861e7B1C57",
+  "0x5a5811E0A22695FEC271fe908E2a1A64Dc3b06F9",
+  "0x254768D47Cf8958a68242ce5AA1aDB401E1feF2B",
+  "0xcfBf34d385EA2d5Eb947063b67eA226dcDA3DC38",
+  "0x99Fc221482ca78664d0288FfC6531cb6dfEB0c5a",
+  "0xA0EA34448738357e0c2e58147b5719A19022ac76",
+]
```
Comment (sweetmantech) [inline hooks/useCreateData.tsx]: can you move this list to the `const` file?

File: hooks/useZoraPremint.tsx [hunk 3]
```diff
@@ -15,44 +15,95 @@ const useZoraPremint = () => {
   const { sendTransaction: sendTxByWallet } = useWalletSendTransaction()
   const { connectWallet } = usePrivy()
 
+  const IS_TESTNET = process.env.NEXT_PUBLIC_TESTNET === "true"
+
+  const proofs = {
```
Comment (sweetmantech) [inline hooks/useZoraPremint.tsx]: can you please move this into a standalone `json` file?
maybe something like `/lib/whitelistProof.json`

File: hooks/useZoraPremint.tsx [hunk 4]
```diff
@@ -15,44 +15,95 @@ const useZoraPremint = () => {
   const { sendTransaction: sendTxByWallet } = useWalletSendTransaction()
   const { connectWallet } = usePrivy()
 
+  const IS_TESTNET = process.env.NEXT_PUBLIC_TESTNET === "true"
```
Comment (sweetmantech) [inline hooks/useZoraPremint.tsx]: IS_TESTNET should already be exported from `consts.tsx`
- if it's not there already, can you move this declaration to the `consts.tsx` file?

File: hooks/useZoraPremint.tsx [hunk 5]
```diff
@@ -15,44 +15,95 @@ const useZoraPremint = () => {
...
-      const SWEETMAN_PROOF = "0x79ae2e7ba528b36d9095a66991f47b7f3ac877be97dda6d556bc14baa58fd48c"
-      const DACIRKUS_PROOF = "0xc747f46b8f122cd0590c39817a019cbc336648ee109c705a235361b3ae9089df"
-      const merkleProof = isSweetman ? SWEETMAN_PROOF : DACIRKUS_PROOF
-      const args = [quantity, presaleMaxQuantity, pricePerToken, [merkleProof], COMMENT]
+      const userProof = proofs[externalWallet.address]
+      console.log(externalWallet.address)
+      if (!userProof || userProof.length === 0) {
+        toast.error("No proof found for address")
+        return { error: true }
+      }
```
Comment (sweetmantech) [inline hooks/useZoraPremint.tsx]: can you move this to a lib?
- `lib/getWhitelistProof.tsx`
- in general - keep files as small as possible. keep code modular.
- HTML in `components`
- functions in `lib`
- state in `hooks` / `providers`
- keep functions small

File: hooks/useZoraPremint.tsx [hunk 6]
```diff
@@ -15,44 +15,95 @@ const useZoraPremint = () => {
...
+
       const response = await publicClient.waitForTransactionReceipt({
         hash: hash as Address,
       })
+
       toast.success("Collected!")
       return getTokenId(response)
     } catch (err) {
       handleTxError(err)
+      toast.error("Transaction rejected")
```
Comment (sweetmantech) [inline hooks/useZoraPremint.tsx]: toast.error already happens in `handleTxError` right?
- why toast a second time?

File: hooks/useCreateData.tsx (hunk 1 above)
Comment (TahirHameed74) [reply hooks/useCreateData.tsx]: Added this comment changes to the PR.

File: hooks/useCreateData.tsx (hunk 2 above)
Comment (TahirHameed74) [reply hooks/useCreateData.tsx]: Added this comment changes to the PR.

File: hooks/useZoraPremint.tsx (hunk 3 above)
Comment (TahirHameed74) [reply hooks/useZoraPremint.tsx]: Added this comment changes to the PR.

File: hooks/useZoraPremint.tsx (hunk 4 above)
Comment (TahirHameed74) [reply hooks/useZoraPremint.tsx]: Added this comment changes to the PR.

File: hooks/useZoraPremint.tsx (hunk 5 above)
Comment (TahirHameed74) [reply hooks/useZoraPremint.tsx]: Added this comment changes to the PR.

File: hooks/useZoraPremint.tsx (hunk 6 above)
Comment (TahirHameed74) [reply hooks/useZoraPremint.tsx]: Added this comment changes to the PR.

File: hooks/useCreateData.tsx [hunk 7]
```diff
@@ -12,12 +13,15 @@ import handleTxError from "@/lib/handleTxError"
 import { useSheetRenderer } from "@/providers/SheetRendererProvider"
 import getTotalSupply from "@/lib/getTotalSupply"
 import usePreparePrivyWallet from "./usePrepareWallet"
+import { whitelistedAddresses } from "@/lib/whitelistAddress"
+import useConnectedWallet from "@/hooks/useConnectedWallet" // Importing your custom hook
```
Comment (sweetmantech) [inline hooks/useCreateData.tsx]: remove comments 🚮 
```suggestion
import useConnectedWallet from "@/hooks/useConnectedWallet"
```

File: hooks/useCreateData.tsx [hunk 8]
```diff
@@ -12,12 +13,15 @@ import handleTxError from "@/lib/handleTxError"
 import { useSheetRenderer } from "@/providers/SheetRendererProvider"
 import getTotalSupply from "@/lib/getTotalSupply"
 import usePreparePrivyWallet from "./usePrepareWallet"
+import { whitelistedAddresses } from "@/lib/whitelistAddress"
```
Comment (sweetmantech) [inline hooks/useCreateData.tsx]: are there multiple addresses here?
if yes, please rename to `whitelistedAddresses` (plural)
```suggestion
import { whitelistedAddresses } from "@/lib/whitelistedAddresses"
```

File: hooks/useCreateData.tsx [hunk 9]
```diff
@@ -33,10 +37,21 @@ const useCreateData = () => {
...
+  const { externalWallet } = useConnectedWallet()
+
+  useEffect(() => {
+    const checkIfWhitelisted = () => {
+      const lowercaseAddress = externalWallet?.address
+      const isWhitelistedAddress = whitelistedAddresses.includes(lowercaseAddress)
+      setIsWhitelisted(isWhitelistedAddress)
+    }
+    checkIfWhitelisted()
+  }, [externalWallet])
```
Comment (sweetmantech) [inline hooks/useCreateData.tsx]: can you move this logic to a new hook?
- `hooks/useWhitelist`
- returns the `isWhitelisted` variable for easy import into any part of the application

File: hooks/useCreateData.tsx [hunk 10]
```diff
@@ -12,12 +13,14 @@ import handleTxError from "@/lib/handleTxError"
...
 import getTotalSupply from "@/lib/getTotalSupply"
 import usePreparePrivyWallet from "./usePrepareWallet"
+import useWhitelist from "./useWhiteList"
 
 const useCreateData = () => {
   const [currentStep, setCurrentStep] = useState(STEPS.CHOOSE_CHARACTER_TYPE)
   const [characterType, setCharacterType] = useState(1)
   const [mintedTokenId, setMintedTokenId] = useState(1)
   const [quantity, setQuantity] = useState(1)
+  // const [isWhitelisted, setIsWhitelisted] = useState(false)
```
Comment (sweetmantech) [inline hooks/useCreateData.tsx]: remove unused code 🚮 
```suggestion
```

File: hooks/useWhiteList.tsx [hunk 11]
```diff
@@ -0,0 +1,25 @@
+import { useState, useEffect } from "react"
+import { whitelistedAddresses } from "@/lib/whitelistAddresses"
+import useConnectedWallet from "@/hooks/useConnectedWallet"
+const useWhitelist = () => {
+  const { externalWallet } = useConnectedWallet()
+  const [isWhitelisted, setIsWhitelisted] = useState(false)
```
Comment (sweetmantech) [inline hooks/useWhiteList.tsx]: is this state being used?
- if no, please remove it

File: hooks/useWhiteList.tsx [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+
+  useEffect(() => {
+    checkIfWhitelisted()
+    if (externalWallet?.address) {
```
Comment (sweetmantech) [inline hooks/useWhiteList.tsx]: why have this check if you're already calling `checkIfWhitelisted`  prior on line 9?

File: hooks/useWhiteList.tsx [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+  useEffect(() => {
+    checkIfWhitelisted()
+  }, [externalWallet])
+
+  const checkIfWhitelisted = () => {
+    const address = externalWallet?.address
+    const isWhitelistedAddress = whitelistedAddresses.includes(address)
+    const istrue = isWhitelistedAddress
+    return istrue
+  }
```
Comment (sweetmantech) [inline hooks/useWhiteList.tsx]: move this to a lib `lib/checkIsWhitelisted`

should remove the need for a hook altogether, right?

**Review by sweetmantech — APPROVED** [review]

**Pull request merged**